### ISSUE TITLE
`depends_on`   causes a syntax problem

### ISSUE THREAD

**Issue by Justin-DynamicD** [hashicorp/vscode-terraform#245]
sample snippet:

```
resource "azurerm_app_service_certificate" "APICERT" {
  name                  = "${var.ssl_keyname}"
  resource_group_name   = "${azurerm_resource_group.RG.name}"
  location              = "${azurerm_resource_group.RG.location}"
  key_vault_secret_id   = "${data.azurerm_key_vault_secret.CERT.id}"
  depends_on            = [azurerm_key_vault_access_policy.APIAccessPolicy]
}
```

In the above sample snippet, the `depends_on` attribute will error on the syntax,citing:

>unexpected token while parsing list: IDENT

This is a functional configuration, so I would expect the extension to not claim an error

**Comment by rowleyaj** [comment]
@Justin-DynamicD what version of terraform and the vscode extension are you using?

**Comment by Justin-DynamicD** [comment]
That would be good to share, huh? ;)

I'm currently using vscode: 1.41.0, extension: Terraform 1.4.0

Slight amendment: I can suppress the IDENT error by using quotes around the target like so:

```
resource "azurerm_app_service_certificate" "APICERT" {
  name                  = "${var.ssl_keyname}"
  resource_group_name   = "${azurerm_resource_group.RG.name}"
  location              = "${azurerm_resource_group.RG.location}"
  key_vault_secret_id   = "${data.azurerm_key_vault_secret.CERT.id}"
  depends_on            = ["azurerm_key_vault_access_policy.APIAccessPolicy"]
}
```

That said, the "quote trick" doesn't work everywhere.  In a for_each block I _also_ see the error, but I've not been able to put things in quotes in a way to supress it.  Take this drive definition from a VM module:

```
dynamic "disk" {
    for_each = [for s in var.VM-Disks: {
      label = contains(keys(s),"label") ? s.label : s.unit_number == "0" ? "${var.VM-NamePrefix}${format("%02d", count.index + 1)}.vmdk" : "${var.VM-NamePrefix}${format("%02d", count.index + 1)}_${s.unit_number}.vmdk"
      unit_number = s.unit_number
      size = contains(keys(s),"size") ? s.size : "${data.vsphere_virtual_machine.datatemplate.disks.0.size}"
      thin_provisioned = contains(keys(s),"thin_provisioned") ? s.thin_provisioned : "${data.vsphere_virtual_machine.datatemplate.disks.0.thin_provisioned}"
    }]
    content {
      label = disk.value.label
      unit_number = disk.value.unit_number
      size = disk.value.size
      thin_provisioned = disk.value.thin_provisioned
    }
  }
```

Here I will get the same error right after the for_each.

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.
